### PR TITLE
feat: Add API to add OAuth token to user model

### LINKED ISSUE #35: Create third party token integration API
## Type of PR

- [x] Feature Request
- [ ] Report Bug
- [ ] Documentation Update
- [ ] Other

## Context

We are planning to handle OAuth Authenticate on the client-side. So we need an API to save the token in our data in our databases. This issue will provide the API for it.

## Check

- [x] I have followed issue creation guidelines
- [x] I have attached proper label with the issue
- [x] The issue is assigned to a assignee

### PR DESCRIPTION
**Pull request by harish-aka-shivi**
# Pull Request Template

## Description

This issue makes an API which will be called by client end to store the OAuth token
Also, this branch is sub-branch of another branch which is yet to be merged. So that is why it has two commits.

Fixes #35 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Test
- [ ] Integration Test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/src/__tests__/userRoute.test.js [hunk 1]
```diff
@@ -67,3 +68,30 @@ describe('User authentication flow', () => {
...
+    const resAddAuthToken = await request(app)
+      .post('/user/add-token')
+      .set('Authorization', `Bearer ${token}`)
+      .send({
+        service: 'github',
+        token: GITHUB_TEST_AUTH_TOKEN,
+      });
+    expect(resAddAuthToken.statusCode).toEqual(200);
+
+    const resDelete = await request(app)
```
Comment (rishabh1403) [inline backend/src/__tests__/userRoute.test.js]: a better place to clean all data would be `afterall` method

File: backend/src/controllers/serviceControllers.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+const getAppropriateService = serviceName => {
+  if (serviceName === 'github') {
+    return githubService;
+  }
+  return githubService;
+};
+
+const getServices = (req, res) => {
+  try {
+    const services = ['github'];
```
Comment (rishabh1403) [inline backend/src/controllers/serviceControllers.js]: why try/catch here ?it's just an assignment

Comment (harish-aka-shivi) [reply backend/src/controllers/serviceControllers.js]: This file change is part of [this PR](https://github.com/pesto-students/batch-11-mernZapp/pull/34/files), so I have made requested change in that PR. So once that PR is merged, I will rebase it and that would reflect the changes.

File: backend/src/__tests__/userRoute.test.js (hunk 1 above)
Comment (harish-aka-shivi) [reply backend/src/__tests__/userRoute.test.js]: Yes, I started with a simple test which worked fine with this technique. But as the tests are growing, it's not scaling well, I need to start using the hook methods.

File: backend/src/index.js [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 app.use(userRouter);
+app.use(servicesRouter);
 
 
 initDb().then(() => {
-  app.listen(3000, () => {
-    console.log('App listening on port 3000!');
-  });
+  if (NODE_ENV !== 'test') {
+    app.listen(3000, () => {
```
Comment (nimish-gupta) [inline backend/src/index.js]: use env variable for the port.

File: backend/src/controllers/userController.js [hunk 4]
```diff
@@ -42,9 +44,21 @@ const deleteUser = (req, res) => {
   }
 };
 
+const addToken = (req, res) => {
+  try {
+    const { service, token } = req.body;
+    req.user.saveServiceToken(service, token);
```
Comment (nimish-gupta) [inline backend/src/controllers/userController.js]: I think you have to wait here.

**Pull request merged**